content/api/commands/origin.md

```diff
@@ -0,0 +1,467 @@
+```
+
+### Arguments
+
+**<Icon name="angle-right"></Icon> url** **_(String)_**
+
+A URL specifying the secondary origin in which the callback is to be executed.
+This should at the very least contain a hostname, and may also include the
+protocol, port number & path. This argument will be used in two ways.
+
+Firstly, it uniquely identifies a secondary origin in which the commands in the
+callback will be executed. The test-runner will inject the Cypress runtime into
+this origin, and then send it code to evaluate in that origin, without violating
+the browser's same-origin policy.
+
+Secondly, it temporarily overrides the `baseUrl` configured in your
+[global configuration](/guides/references/configuration#Global) whilst inside
+the callback. So `cy.visit()` will navigate relative to this URL, not the
+configured `baseUrl`.
+
+**<Icon name="angle-right"></Icon> options** **_(Object)_**
+
+Pass in an options object to control the behavior of `cy.origin()`.
+
+| option | description                                                                                                                                                                                                    |
+| ------ | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| args   | Plain JavaScript object which will be serialized and sent from the primary origin to the secondary origin, where it will be deserialized and passed into the callback function as its first and only argument. |
+
+<Alert type="warning">
+
+The `args` object is the **only** mechanism via which data may be injected into
+the callback, the callback is **not** a closure and does not retain access to
+the JavaScript context in which it was declared.
+
+</Alert>
+
+**<Icon name="angle-right"></Icon> callbackFn** **_(Function)_**
+
+The function containing the commands to be executed in the secondary origin.
+
+This function will be stringified, sent to the Cypress instance in the secondary
+origin and evaluated. If the `args` option is specified, the deserialized args
+object will be passed into the function as its first and only argument.
+
+There are a number of limitations placed on commands run inside the callback,
+please see [Callback restrictions](#Callback-restrictions) section below for a
+full list.
+
+The function's return value is ignored. ???
+
+### Yields [<Icon name="question-circle"/>](/guides/core-concepts/introduction-to-cypress#Subject-Management)
+
+- `cy.origin()` yields `null`.
+- `cy.origin()` cannot be chained further.
+
+## Examples
+
+### Using dynamic data in a secondary origin
+
+Callbacks are executed inside an entirely separate instance of Cypress, so
+arguments must be transmitted to the other instance by means of serialization &
+deserialization. The interface for this mechanism is the `args` option.
```

Comment:
Specifically we use the structured clone algorithm, if you want to delve that deep.

https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API/Structured_clone_algorithm